### PR TITLE
Bugfix for classwise wrapper compute logic

### LINKED ISSUE #1220: __call__ using ClassWrapper does not update wrapped metric
## 🐛 Bug

Using __call__ with ClasswiseWrapper does not update the Metric within, such that subsequent calls to compute are only using the last batch input.

### To Reproduce

See code sample

#### Code sample

```py
import numpy as np
import torch
import torchmetrics

np.random.seed(42)

m1 = torchmetrics.Precision(num_classes=3, average=None)
m2 = torchmetrics.Precision(num_classes=3, average=None)
mcw1 = torchmetrics.ClasswiseWrapper(torchmetrics.Precision(num_classes=3, average=None))
mcw2 = torchmetrics.ClasswiseWrapper(torchmetrics.Precision(num_classes=3, average=None))

for i in range(2):
    print(f"Epoch: {i}")
    preds = torch.Tensor(np.random.uniform(0, 1, (100, 3)))
    targets = torch.Tensor(np.random.randint(0, 2, (100, 3))).int()
    m1(preds, targets)
    m2.update(preds, targets)
    mcw1(preds, targets)
    mcw2.update(preds, targets)
    print(f"Prec:             {m1.compute().numpy()}")
    print(f"Prec update:      {m2.compute().numpy()}")
    print(f"Prec wrap:        {np.asarray([v.numpy() for v in mcw1.compute().values()])}")
    print(f"Prec wrap update: {np.asarray([v.numpy() for v in mcw2.compute().values()])}")
```

Output is:

```
Epoch: 0
Prec:             [0.42857143 0.537037   0.509434  ]
Prec update:      [0.42857143 0.537037   0.509434  ]
Prec wrap:        [0.42857143 0.537037   0.509434  ]
Prec wrap update: [0.42857143 0.537037   0.509434  ]
Epoch: 1
Prec:             [0.46464646 0.5247525  0.48039216]
Prec update:      [0.46464646 0.5247525  0.48039216]
Prec wrap:        [0.5       0.5106383 0.4489796]
Prec wrap update: [0.46464646 0.5247525  0.48039216]
```

### Expected behavior

The values should be the same whether calling `update` or `__call__`

### Environment

- TorchMetrics version: 0.9.3
- Python & PyTorch Version (e.g., 1.0): 1.12.0
- Any other relevant information such as OS (e.g., Linux): Ubuntu 20.04

### PR DESCRIPTION
**Pull request by SkafteNicki**
## What does this PR do?

Fixes #1220 
Some of the logic in `_wrap_update` and `_wrap_compute` was messing with the `compute` method in the `ClasswiseWrapper`. This PR makes them empty wrappers for that metric and also adds relevant testing.

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to **update the docs**?
- [x] Did you write any new **necessary tests**?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by Borda — APPROVED** [review]

File: src/torchmetrics/wrappers/classwise.py [hunk 1]
```diff
@@ -76,3 +79,21 @@ def compute(self) -> Dict[str, Tensor]:
...
+        @functools.wraps(update)
+        def wrapped_func(*args: Any, **kwargs: Any) -> None:
+            update(*args, **kwargs)
+
+        return wrapped_func
+
+    def _wrap_compute(self, compute: Callable) -> Callable:
+        """Overwrite to do nothing."""
+
+        @functools.wraps(compute)
```
Comment (justusschock) [inline src/torchmetrics/wrappers/classwise.py:95]: couldn't you just return the original compute method here?

File: src/torchmetrics/wrappers/classwise.py [hunk 2]
```diff
@@ -76,3 +79,21 @@ def compute(self) -> Dict[str, Tensor]:
 
     def reset(self) -> None:
         self.metric.reset()
+
+    def _wrap_update(self, update: Callable) -> Callable:
+        """Overwrite to do nothing."""
+
+        @functools.wraps(update)
```
Comment (justusschock) [inline src/torchmetrics/wrappers/classwise.py:86]: couldn't you just return the original update here?

**Review by justusschock — APPROVED** [review]

File: src/torchmetrics/wrappers/classwise.py (hunk 1 above)
Comment (SkafteNicki) [reply src/torchmetrics/wrappers/classwise.py]: yes, of cause :)
